src/modules/blob.js

```diff
@@ -79,7 +80,7 @@ class BlobModule {
       // eslint-disable-next-line no-param-reassign
       signature = getSignatureFromKeyringPair(keyPair, serializedBlob);
     }
-    return this.module.new(blob, signature.toJSON());
+    return await this.sendTransaction(this.module.new(blob, signature.toJSON()), false);
```

Comment:
This means that the transaction once created has to sent. We *must* have a way to create a transaction but not send (was mentioned in the corresponding issue) as that's needed to estimate the fees required (using `paymentInfo`) and sending transactions in parallel (using `utility.batch`). I don't mind if you accomplish that by having a `send` argument (in `new`) that's true by default or have auxiliary function like `writeNew`.

Reply:
i was thinking that in this case user could hook the custom sign tx's in, it is hard to think of a good solution for that without seeing a use case where the tx is separated from sending

Reply:
at this point im leaning towards a new argument in new that is true by default, otherwise return the tx

Reply:
Its fine if you are short on time but keep in mind the return type becomes ugly now.

Reply:
yeah on further thought not so happy with that, would rather make a new method to create tx and keep current ones using sendtransaction. see https://github.com/docknetwork/sdk/pull/138/commits/64df79b14625068a21db1ebd791d0d305b8f9683